config/application.rb

```diff
@@ -272,6 +282,21 @@ def watchable_args
     Sprockets.register_mime_type 'application/javascript', extensions: ['.js', '.es6', '.js.es6'], charset: :unicode
     Sprockets.register_postprocessor 'application/javascript', DiscourseJsProcessor
 
+    # The sprockets sourcemap pipeline overrites our ember-cli-generated sourcemaps, so let's disable it
+    # (it doesn't work with our DiscourseJsProcessor anyway)
+    config.assets.configure do |env|
+      env.cache = nil # TODO REMOVE THIS
+      env.unregister_processor('application/javascript', Sprockets::Preprocessors::DefaultSourceMap)
+      env.unregister_postprocessor('application/javascript', Sprockets::Rails::SourcemappingUrlProcessor)
+    end
+
+    if EmberCli.enabled?
+      Discourse::Application.initializer :prepend_ember_assets do |app|
+        # Needs to be in its own initializer so it runs after Sprockets' append_assets_path
```

Comment:
Is `append_assets_path` an initializer defined in sprockets?

Reply:
Yes. I'll make the comment clearer :+1: